libs/build.nix

```diff
@@ -57,6 +63,149 @@ _:
+              "${config.packages.lbf-prelude-haskell}"
+            ];
+
+            modules = [
+              (_: {
+                packages = {
+                  allComponent.doHoogle = true;
+                  allComponent.doHaddock = true;
+
+                  # Enable strict compilation
+                  lbt-plutus-haskell.configureFlags = [ "-f-dev" ];
+                };
+              })
+            ];
+
+            shell = {
+
+              withHoogle = true;
+
+              exactDeps = true;
+
+              nativeBuildInputs = config.settings.shell.tools
+                ++ [ config.packages.lbf-prelude-to-haskell ];
+
+              # Note: the `additional` (contrast to `packages`) attribute
+              # includes the dependencies + the package itself. See:
+              # https://input-output-hk.github.io/haskell.nix/reference/library.html#shellfor
+              # This *must* be the name of the autogenerated cabal package from
+              # `lbf-prelude-haskell`
+              additional = ps: [ ps.lbf-prelude ];
+
+              tools = {
+                cabal = { };
+                haskell-language-server = { };
+              };
+
+              shellHook = lib.mkForce config.settings.shell.hook;
+            };
+          };
+          hsNixFlake = (pkgs.haskell-nix.cabalProject' [
+            inputs.mlabs-tooling.lib.mkHackageMod
+            inputs.mlabs-tooling.lib.moduleMod
+            project
+          ]).flake { };
+        in
+        hsNixFlake.devShell;
+
+      dev-plutustx =
+        # Note:
+        # Similarly to `dev-prelude-haskell`, `packages.lbf-plutus-haskell`
+        # essentially generates a cabal project from the `*.lbf` schemas; and
+        # the following uses `haskell.nix` to convert the `.cabal` project into
+        # a dev shell.
+        # This is a dev shell which provides ghc with
+        #   - `lbf-plutus-haskell` package (and its dependencies)
+        #   - the CLI application (`lbf-plutus-to-haskell`) to compile `.lbf`
+        #   schemas.
+        #
+        # Note:
+        # This is mostly duplicated code from `dev-prelude-haskell`
+        let
+          project = { lib, ... }: {
+            src = config.packages.lbf-plutus-haskell;
+
+            name = "lbf-plutus-haskell";
+
+            inherit (config.settings.haskell) index-state compiler-nix-name;
+
+            extraHackage = [
+              "${config.packages.lbr-prelude-haskell-src}"
+              "${config.packages.lbf-prelude-haskell}"
+              "${config.packages.lbr-plutus-haskell-src}"
+              "${config.packages.lbf-plutus-haskell}"
+            ];
+
+            modules = [
+              (_: {
+                packages = {
+                  allComponent.doHoogle = true;
+                  allComponent.doHaddock = true;
+
+                  # Enable strict compilation
+                  lbt-plutus-haskell.configureFlags = [ "-f-dev" ];
```

Comment:
Also missed the name here, `lbf-plutus`, but I'm not entirely sure we need this whole `modules` section tbh. Well it's useful to also generated docs and hoogle, but strict compilation will definitely fail on autogenerated code x)